### PR TITLE
Adding AWS CLI for COS upload to support python 3.9

### PR DESCRIPTION
**Pull request by AmarnatReddy**
# Description
Adding Aspera SDK for python 3.9 support

Logs : 
https://159.23.92.24/job/test_aspera_transfer/

I have tried executing the script in different threads . it has created 2 instances of aspera server

```
[root@cephci-rhel9-agent-02 ~]# netstat -tlnp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN      1/systemd           
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      30797/sshd: /usr/sb 
tcp        0      0 127.0.0.1:34525         0.0.0.0:*               LISTEN      629079/./linux-amd6 
tcp        0      0 127.0.0.1:35631         0.0.0.0:*               LISTEN      629106/./linux-amd6 
tcp        0      0 127.0.0.1:61981         0.0.0.0:*               LISTEN      629079/./linux-amd6 
tcp        0      0 127.0.0.1:53658         0.0.0.0:*               LISTEN      629106/./linux-amd6 
tcp6       0      0 :::111                  :::*                    LISTEN      1/systemd           
tcp6       0      0 :::22                   :::*                    LISTEN      30797/sshd: /usr/sb 
```

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Review by tintumathew10 — COMMENTED** [review]
Tested it. LGTM

File: storage/ibm_cos_aspera_sdk.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+        self.endpoint = self._conf["endpoint"]
+        self.aspera_url = "https://download.asperasoft.com/download/sw/sdk/transfer/1.1.1/linux-amd64-52a85ef.tar.gz"
+        self.port = "55009"
+
+    def is_port_in_use(self, port):
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            return s.connect_ex(("localhost", port)) == 0
+
+    def terminate_aspera(self):
```
Comment (sunilkumarn417) [inline storage/ibm_cos_aspera_sdk.py:27]: what if another process using this port for transfer? or needs aspera sdk

File: storage/ibm_cos_aspera_sdk.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            "transfer_pb2_grpc.py"
+        ):
+            command = "cp linux-amd64/connectors/python/transfer_pb2* ."
+            subprocess.run(command, shell=True)
+
+    def upload_directory(self, bucket_name, local_dir, object_prefix):
+        """
+        upload the directory to IBM COS to given bucket name
+        Args:
+            bucket_name:
```
Comment (sunilkumarn417) [inline storage/ibm_cos_aspera_sdk.py:145]: add args definition

File: storage/ibm_cos_aspera_sdk.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        output = output.decode("utf-8").split("\n")
+        output = [
+            line.split()
+            for line in output
+            if line.startswith("tcp") or line.startswith("udp")
+        ]
+        port_numbers = [int(line[3].split(":")[-1]) for line in output]
+        port_numbers = sorted(list(set(port_numbers)))
+        if self.port in port_numbers:
+            self.port = int("5500" + str(random.randint(0, 99999)).zfill(2))
```
Comment (sunilkumarn417) [inline storage/ibm_cos_aspera_sdk.py:53]: lets stick to port range (49153, 65535).

**Comment by AmarnatReddy** [comment]
I have moved to awscli instead of aspera.
It has support for all the python releases 

Pass Log: 
https://jenkins.ceph.redhat.com/view/RHCS%20QE/job/rhceph-unsigned-compose-listener_amk/18/console

**Comment by sunilkumarn417** [comment]
> I have moved to awscli instead of aspera. It has support for all the python releases Ref : https://github.com/aws/aws-cli
> 
> Pass Log: https://jenkins.ceph.redhat.com/view/RHCS%20QE/job/rhceph-unsigned-compose-listener_amk/18/console

Thanks Amar for finding out the alternative.

Let's change the description and add test results there.

**Review by sunilkumarn417 — APPROVED** [review]

**Pull request merged**